examples/PSyclone/psyclone.py

```diff
@@ -60,29 +62,34 @@ def test_function(options, psy):
                 print dir(my_psy.invokes)
                 my_invoke = my_psy.invokes.invoke_map[invoke.name]
                 print my_invoke.name
-                from transformations import GOceanLoopFuseTrans, TransformationError
+                from transformations import GOceanLoopFuseTrans, \
+                    TransformationError
+                trans = GOceanLoopFuseTrans()
                 for loops in values:
-                    psy, _ = trans.apply(loops[0], loops[1])
+                    _, _ = trans.apply(loops[0], loops[1])
             my_invoke.view()
     exit(1)
     return True, [42]
 
 
 class ArrayBounds(Choice):
+    '''A specialisation of the melody Choice input class providing
+    array-bound-specific choices for the GOcean 1.0 api.'''
     def __init__(self):
-        Choice.__init(self, name="Array Bounds",
-                      options=["", "Specified", "Constant"])
+        Choice.__init__(self, name="Array Bounds",
+                        options=["", "Specified", "Constant"])
 
 
 class ModuleInline(Subsets):
-    ''' '''
+    '''A specialisation of the melody Subsets input class providing all
+    possible combinations of module inlining for the particular code.'''
 
     def __init__(self, psy=None):
         self._psy = psy
         # get the kernels that can be inlined from PSyclone
         kernels = []
         kernel_names = []
-        for invoke in psy.invokes.invoke_list:
+        for invoke in self._psy.invokes.invoke_list:
```

Comment:
Quite right. I've made psy a required argument. Note this code is all subject to change as it is not working correctly at the moment so I've not bothered too much about being graceful with errors.